[DOCS] Add OSSM creating resources in RHEL 10 page (#105479)
openshift/openshift-docs: modules/ossm-creating-resources-in-rhel10-system.adoc

// Module included in the following assemblies:
// install/ossm-installing-openshift-service-mesh.adoc

:_mod-docs-content-type: PROCEDURE
[id="ossm-creating-resources_in_rhel10_system_{context}"]

> = Creating the Istio and IstioCNI resources in RHEL 10 systems

Reviewer: I am actually thinking we should pause publishing this documentation, because RHEL10 nodes are not yet supported as GA features with OpenShift, so we don't want to imply they are supported with OSSM just yet. The time to publish this will be the first release that supports OCP 5 (3.4 is the earliest likely). 

With that said, I also don't think it is a good experience that there is a separate install procedure for RHEL10 after the main install procedure - I suspect many people will miss it. I think it would be better to include the config differences within each step (e.g. Install Istio CNI with an example for iptables and nftables, then Install Istio with an example for iptables and nftables). I also think there's a need for short explanation on why you would choose nftables.

> = Creating the Istio and IstioCNI resources in RHEL 10 systems

Reviewer 2: > The time to publish this will be the first release that supports OCP 5 (3.4 is the earliest likely).

I'm fine if we want to publish it for OCP 5, but let me share the current situation.

Starting with OCP 4.18, OVN-Kubernetes already uses nftables as the default backend. Although nftables is mandatory on RHEL 10, it is supported on RHEL 9 aswell. 
From a technical standpoint, the nftables implementation in OSSM should work even with RHEL9. The main consideration is whether we have adequate QE bandwidth to complete validation on the existing OCP 4.18+ platforms.

> With that said, I also don't think it is a good experience that there is a separate install procedure for RHEL10 after the main install procedure - I suspect many people will miss it. I think it would be better to include the config differences within each step (e.g. Install Istio CNI with an example for iptables and nftables, then Install Istio with an example for iptables and nftables). I also think there's a need for short explanation on why you would choose nftables.

Totally agree. I had similar thoughts when I started reviewing the PR.

> Enable native nftables support in {SMProduct} 3.2 for Red Hat Enterprise Linux 10 systems by configuring the `values.global.nativeNftables` field in the `{istio}` and `IstioCNI` resources.

Reviewer 2: >... native nftables support in {SMProduct} 3.2 for Red Hat Enterprise Linux 10 systems

Do we want to expand the scope further and include **RHCOS nodes based on RHEL 10**?

// Module included in the following assemblies:
// install/ossm-installing-openshift-service-mesh.adoc

:_mod-docs-content-type: PROCEDURE
[id="ossm-creating-resources_in_rhel10_system_{context}"]
= Creating the Istio and IstioCNI resources in RHEL 10 systems

Nftables is a successor of iptables tool in the Linux kernel. The iptables interface has been deprecated and replaced by the nftables in Red Hat Enterprise Linux 10 (RHEL 10) systems. To deploy {istio} in a RHEL 10 system, you must set a feature flag as `values.global.nativeNftables=true` when you create two resources: `Istio` and `IstioCNI`. This feature was included in the {SMProduct} 3.2.

.Prerequisties

* The {SMProductShortName} Operator must be installed.

* You are logged in to the {ocp-product-title} web console as cluster-admin.

.Procedure

. In the {ocp-product-title} web console, click *Operators* -> *Installed Operators*.

. Select `istio-cni` in the *Project* drop-down menu.

> . Click the {SMProductShortName} Operator.

Reviewer 3: ```suggestion
. Click the *{SMProductShortName} Operator* option.
```
or
```suggestion
. Click *{SMProductShortName} Operator*.
```
If this is an option the user has to select from, I would suggest the above style changes.

. Click *IstioCNI* in the *Provided APIs* column.

. Click the `IstioCNI` instance, named `default`, in the *Name* column.

. Click *YAML* to view the `IstioCNI` configuration and make modifications:
+
.Example configuration
[source,yaml,subs="attributes,verbatim"]
----
apiVersion: sailoperator.io/v1
kind: IstioCNI
metadata:
  name: default
spec:
  namespace: istio-cni
  values:
    global:
      nativeNftables: true
# ...
----

. Click *Create*. This action deploys the Istio CNI plugin.
+
When `State: Healthy` appears in the *Status* column, the Istio CNI plugin is successfully deployed.

. In the {ocp-product-title} web console, click *Operators* -> *Installed Operators*.

. Select `istio-system` in the *Project* drop-down menu.

> . Click the {SMProductShortName} Operator.

Reviewer 3: ```suggestion
. Click the *{SMProductShortName} Operator* option.
```
or
```suggestion
. Click *{SMProductShortName} Operator*.
```
Same comment as above: If this is an option the user has to select from, I would suggest the above style changes.

. Click *Istio* in the *Provided APIs* column.

. Click the `Istio` instance, named `default`, in the *Name* column.

> . Click *YAML* to view the `Istio` configuration and make modifications:

Reviewer 3: ```suggestion
. Click *YAML* to view the `{istio}` configuration and make changes similar to the following example:
```

+
.Example configuration
[source,yaml,subs="attributes,verbatim"]
----
apiVersion: sailoperator.io/v1
kind: Istio
metadata:
  name: default
spec:
  namespace: istio-system
  values:
    global:
      nativeNftables: true
# ...
----

> . Click *Create*. This action deploys the {istio} control plane.

Reviewer 3: ```suggestion
. Click *Create* to deploy the {istio} control plane.
```

+

> When `State: Healthy` appears in the *Status* column, {istio} is successfully deployed.

Reviewer 3: ```suggestion
To verify that the {istio} is successfully deployed, see if the *Status* column shows `State` as `Healthy`.
```
